README.md

```diff
@@ -42,7 +42,346 @@ Or install it yourself as:
+
+```ruby
+puts Lotus::Mailer.configuration.root # => #<Pathname:app/templates>
+InvoiceMailer.templates               # => "{:html => templateObject1; :txt => templateObject2}"
+templateObject1.file                  # => "root/invoice_mailer.html.erb"
+templateObject2.file                  # => "root/invoice_mailer.txt.erb"
+```
+
+Each mailer can specify a different template for a specific format:
+
+```ruby
+module Articles
+  class Create
+    include Lotus::View
+
+    template :haml, 'invoice.haml.erb'
+  end
+end
+
+InvoiceMailer.templates[:haml].file  # => "root/invoice.haml.erb"
+```
+
+### Engines
+
+The builtin rendering engine is [ERb](http://en.wikipedia.org/wiki/ERuby).
+
+### Root
+
+Template lookup is performed under the `Lotus::Mailer.configuration.root` directory. You can specify a different path on a per mailer basis:
+
+```ruby
+class MailerWithDifferentRoot
+  include Lotus::Mailer
+
+  root 'path/to/root'
+end
+```
+
+### Configuration
+
+__Lotus::Mailer__ can be configured with a DSL that determines its behavior.
+It supports a few options:
+
+```ruby
+require 'lotus/mailer'
+
+Lotus::Maler.configure do
+  # Set the root path where to search for templates
+  # Argument: String, Pathname, #to_pathname, defaults to the current directory
+  #
+  root '/path/to/root'
+
+  # Set the Ruby namespace where to lookup for mailers
+  # Argument: Class, Module, String, defaults to Object
+  #
+  namespace 'MyApp::Mailer'
+```
+
+All those global configurations can be overwritten at a finer grained level:
+`mailers`. Each `mailer` has its own copy of the global configuration, so
+that changes are inherited from the top to the bottom, but not bubbled up in the
+opposite direction.
+
+```ruby
+require 'lotus/mailer'
+
+Lotus::Mailer.configure do
+  root '/path/to/root'
+end
+
+class Show
+  include Lotus::Mailer
+  root '/another/root'
+end
+
+Lotus::Mailer.configuration.root # => #<Pathname:/path/to/root>
+Show.root                      # => #<Pathname:/another/root>
+```
+
+### Delivery Details
+
+When a Ruby object includes __Lotus::Mailer__, a developer can customize delivery details, such as `from`, `to` and `subject`.
+
+The `from` field determines the sender of the email. It can accept a string:
+
+```ruby
+class InvoiceMailer
+  include Lotus::Mailer
+
+  from "noreply@example.com"
+end
+```
+
+The `from` field can also accept a `Proc` to determine its value:
+
+```ruby
+class InvoiceMailer
+  include Lotus::Mailer
+
+  from -> { customized_sender }
+
+  def customized_sender
+    "user-#{ user.id }@example.com"
+  end
+end
+```
+
+The `to` field represents one or more emails that will be the recipients of the mail.
+
+It can accept a string:
+
+```ruby
+class InvoiceMailer
+  include Lotus::Mailer
+
+  to "noreply@example.com"
+end
+```
+
+The to field also accept an array of strings:
+
+```ruby
+class InvoiceMailer
+  include Lotus::Mailer
+
+  to "[noreply1@example.com, noreply2@example.com]"
```

Comment:
the library should eventually support valid e-mail addresses. It's pretty complex to parse an email address and i suppose the underlying mail library will support that. In any case: an email address is just a string. So the interface does not change.
Lotus::Mail should probably validate the format and raise if something is wrong.